code/modules/jobs/job_types/gimmick.dm

```diff
@@ -17,7 +17,7 @@
 
 /datum/job/gimmick/New()
 	. = ..()
-	GLOB.civilian_positions |= title
+	GLOB.gimmick_positions |= title
```

Comment:
Filling `GLOB.gimmick_position` on runtime won't be loved by `exp_jobsmap`'s init; `civilian_positions | gimmick_positions` won't update properly and miss gimmick positions because it creates a new list instead of referencing to updating GLOB lists. You'll need `GLOB.gimmick_position` that is initialized just like any other position GLOB list.